typings/styled-components.d.ts

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import { StatelessComponent } from "react";
+
+interface ThemeProps {
+  theme: Object;
+}
+
+type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type StyledProps<P> = P & ThemeProps;
+type Interpolation<P> = ((executionContext: StyledProps<P>) => string) | string | number
+
+interface StyledFunction<T, P> {
+  (strs: TemplateStringsArray, ...fns: Array<Interpolation<P>>): T & { props: StyledProps<P> };
+}
+
+interface StyledInterface {
+  <C extends React.ComponentClass<P>, P, ThemeInterface>(component: ConstrainedProps<C, P>): StyledFunction<C, P>;
+  <C extends React.StatelessComponent<P>, P, ThemeInterface>(component: ConstrainedProps<C, P>): StyledFunction<C, P>;
```

Comment:
ok, @patrick91 I've created a PR to yours: patrick91#3
`native' module hasn't been done yet. Let me know if would like to incorporate my changes.